doc/mgmt/config_replace_rollback_HLD.md

```diff
@@ -0,0 +1,415 @@
+| **Parameters**                                          |
+| source-url                                              | config://filename<br>home://filename<br>ftp://[userid[:passwd]@]{hostname \| host-ip}/file-path<br>http[s]://[userid[:passwd]@]{hostname \| host-ip}/file-path<br>scp://[userid[:passwd]@]{hostname \| host-ip}/file-path|
+| running-configuration                                   | Current running configuration file |
+| replace                                                 | This keyword represents that the configurations in source-url will replace current running configuration  |
+| **Default**                                             | Merge |
+| **Command mode**                                        | EXEC |
+| **Roles**                                               | admin |
+| **Example**                                             | Sonic#copy http://inter128.acme.com/sample.json running-configuration replace<br>Sonic# |
+| **Support**                                             | 4.1 |
+
+#### 3.2.2.2 Configuration Session Mode Commands
+##### 3.2.2.2.1 Replace Command
+| replace command                                         | replace command inside configuration session to replace running configuration  |
+| ------------------------------------------------------- | ------------------------------------------------------------ |
+| **Syntax**                                              | replace source-url  |
+| **Parameters**                                          |
+| source-url                                              | config://filename<br>home://filename<br>ftp://[userid[:passwd]@]{hostname \| host-ip}/file-path<br>http[s]://[userid[:passwd]@]{hostname \| host-ip}/file-path<br>scp://[userid[:passwd]@]{hostname \| host-ip}/file-path|
+| **Default**                                             |  |
+| **Command mode**                                        | Config-s |
+| **Roles**                                               | admin |
+| **Example**                                             | Sonic(config-s)#replace http://inter128.acme.com/sample.json<br>Sonic(config-s)# |
+| **Support**                                             | 4.1 |
+##### 3.2.2.2.2 commit command
+| commit command                                          | commit command inside configuration session to kick start the timer or confirm the configuration changes |
+| ------------------------------------------------------- | ------------------------------------------------------------ |
+| **Syntax**                                              | commit [label \<LABEL\>] [time seconds] \| [confirm]  |
+| **Parameters**                                          |
+| confirm						  | To make the configuration changes permanent in running-configuration |
+| time   						  | 30 .. 3600 seconds for which the configuration changes kept in running-configuration as trial stage, default - 30 |
```

Comment:
- commit without timer ("commit") will immediately apply changes to running-configuration. No timer will be running. Behavior(Behavior that Broadcom is going to support) will be untouched. That's why no default value is mentioned in "Default" section in this table
- commit with timer ("commit time 30") will store snapshot of running configuration, apply changes to running configuration and kick start the timer for 30 minutes. If timer expires, the snapshot will be restored, discarding changes that were committed with timer. If "commit confirm" is executed before the timer expiry, timer will be killed and snapshot will be removed from the system to make the changes permanent in running configuraiton

Reply:
There is a default mentioned on line 277 "30 .. 3600 seconds for ...., default  - 30"  ? What is the meaning of that "default  -  30" ?

Do you mean Dell is going to implement this as "commit time" without a value for the second then it would mean a 30 second (or minute?) timer for the rollback ?  If that is the case the syntax should be "commit [label \<LABEL\>] [time [seconds]] \| [confirm] "

Reply:
The unit is seconds, range in 30 - 3600. There is no default value for the timer. We will fix it.
commit [label <LABEL>] [time seconds] | [confirm]